### PR TITLE
Add support for C

### PR DESCRIPTION
**Pull request by glokta1**

### CODE REVIEW

File: compiled_starters/c/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+![progress-banner](https://codecrafters.io/landing/images/default_progress_banners/shell.png)
+
+This is a starting point for C solutions to the
+["Build Your Own Shell" Challenge](https://app.codecrafters.io/courses/shell/overview).
+
+_In this challenge, you'll build your own POSIX compliant shell that's capable
```
Comment (rohitpaulk) [inline compiled_starters/c/README.md:6]: I'm not sure where the `_` surrounding this paragraph came from - isn't present in the Redis challenge for example: 

Screenshot 2024-06-05 at 11 47 10

@glokta1 can we check other challenges, make sure we're consistent across all? I don't think the italics help, best removed

File: solutions/c/01-oo8/diff/app/main.c.diff [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ #include <stdio.h>
+
+ int main() {
+-  // You can use print statements as follows for debugging, they'll be visible when running tests
+-  printf("Logs from your program will appear here!\n");
+-
+-  // Uncomment this block to pass the first stage
+-  //   printf("$ ");
+-  //   fflush(stdout);
++    printf("$ ");
```
Comment (rohitpaulk) [inline solutions/c/01-oo8/diff/app/main.c.diff:11]: @glokta1 again, not self-reviewing thoroughly enough - see the indentation mismatch here? This is one of the first things a user will see when they use CodeCrafters, and it'll look very amateur to mess up indentation

File: compiled_starters/c/app/main.c [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#include <stdio.h>
+
+int main() {
+  // You can use print statements as follows for debugging, they'll be visible when running tests
+  printf("Logs from your program will appear here!\n");
+
+  // Uncomment this block to pass the first stage
+  //   printf("$ ");
```
Comment (rohitpaulk) [inline compiled_starters/c/app/main.c:8]: Added a note below in the diff, the way we've indented commented code here causes incorrect indentation in the diff. Follow what other challenges do: 

Screenshot 2024-06-05 at 11 48 42

File: compiled_starters/c/README.md (hunk 1 above)
Comment (glokta1) [reply compiled_starters/c/README.md]: @rohitpaulk Yup, without italics everywhere else. Got confused by the formatting in the template README. Fixed.

**Review by rohitpaulk — APPROVED** [review]

**Pull request merged**